src/enclosure.c

```diff
@@ -171,3 +184,151 @@ int enclosure_open(const struct enclosure_device *enclosure)
 
 	return fd;
 }
+
+static struct enclosure_device *find_enclosure(char *enclosure_id)
+{
+	struct enclosure_device *enclosure_device;
+	list_for_each(sysfs_get_enclosure_devices(), enclosure_device) {
+		if (strncmp(enclosure_device->dev_path, enclosure_id, PATH_MAX) == 0) {
+			return enclosure_device;
+		}
+	}
+	return NULL;
+}
+
+static struct ses_slot *find_enclosure_slot_by_index(struct enclosure_device *encl, int index)
+{
+	for (int i = 0; i < encl->slots_count; i++) {
+		if (encl->slots[i].index == index) {
+			return &encl->slots[i];
+		}
+	}
+	return NULL;
+}
+
+static status_t _enclosure_get_slot(struct enclosure_device *encl, int index,
+									const char *device, struct slot_response *slot_res)
```

Comment:
too many tabs here?

Reply:
> too many tabs here?

My editor is setup for 4 spaces per tab, but apparently the expectation is 8?

Reply:
yes :)

Reply:
Done